### PR TITLE
[REF] test_pylint: Add reporter to show just the message

### PR DESCRIPTION
**Pull request by moylop260**
Before this PR:
  - screen shot 2017-07-14 at 22 10 34

After this PR:
  - screen shot 2017-07-14 at 22 01 54

### CODE REVIEW

File: odoo/addons/test_pylint/tests/test_pylint.py [hunk 1]
```diff
@@ -115,11 +124,8 @@ def test_pylint(self):
...
-            process = subprocess.Popen(['pylint'] + options + paths, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        except (OSError, IOError):
-            self._skip_test('pylint executable not found in the path')
-        else:
-            out, err = process.communicate()
-            if process.returncode:
-                self.fail("\n" + out + "\n" + err)
+        reporter = Reporter()
+        lint_res = lint.Run(options + paths, reporter=reporter, exit=False)
+        lint_stats = lint_res.linter.stats
```
Comment (KangOl) [inline odoo/addons/test_pylint/tests/test_pylint.py]: What the purpose of this variable?

File: odoo/addons/test_pylint/tests/test_pylint.py [hunk 2]
```diff
@@ -1,24 +1,47 @@
...
 from os.path import join
 
 from odoo.tests.common import TransactionCase
 from odoo import tools
 from odoo.modules import get_modules, get_module_path
 
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
```
Comment (xmo-odoo) [inline odoo/addons/test_pylint/tests/test_pylint.py]: Is there a reason for not using io everywhere? Also beware, io.BytesIO and io.StringIO are *much* stricter about their values than the old StringIO.StringIO is, even in Python 2 giving bytes to io.StringIO or text to io.BytesIO will blow up.

File: odoo/addons/test_pylint/tests/test_pylint.py [hunk 3]
```diff
@@ -115,11 +138,11 @@ def test_pylint(self):
...
 
-        try:
-            process = subprocess.Popen(['pylint'] + options + paths, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        except (OSError, IOError):
-            self._skip_test('pylint executable not found in the path')
-        else:
-            out, err = process.communicate()
-            if process.returncode:
-                self.fail("\n" + out + "\n" + err)
+        with _patch_streams(StringIO()):  # Avoid show 'no config file msg'
```
Comment (xmo-odoo) [inline odoo/addons/test_pylint/tests/test_pylint.py]: This should not be necessary [if you use the `pylint.lint.PyLinter` class directly](https://github.com/odoo/odoo-extra/pull/94/files#diff-2a21db3452219209e5224d39432c7c1dR34), though you will have to perform the actions of the various options.

**Comment by moylop260** [comment]
https://github.com/odoo/odoo/pull/27313

**Pull request closed without merging**